--- graphics/exiv2/Portfile.orig	2021-11-21 17:51:01.000000000 +0100
+++ graphics/exiv2/Portfile	2021-11-21 17:51:04.000000000 +0100
@@ -41,6 +41,7 @@
 patchfiles-append   patch-remove-no-format-overflow.diff
 
 configure.args-append \
+                    -DEXIV2_ENABLE_BMFF=On \
                     -DIconv_INCLUDE_DIR=${prefix}/include \
                     -DIconv_LIBRARY=${prefix}/lib/libiconv.dylib \
                     -DZLIB_ROOT=${prefix}
