--- gettext-runtime/intl/lock.c.orig	2010-06-06 16:49:57.000000000 +0400
+++ gettext-runtime/intl/lock.c	2012-08-24 20:29:00.000000000 +0400
@@ -280,7 +280,7 @@
 
 # if HAVE_PTHREAD_MUTEX_RECURSIVE
 
-#  if defined PTHREAD_RECURSIVE_MUTEX_INITIALIZER || defined PTHREAD_RECURSIVE_MUTEX_INITIALIZER_NP
+#  if !(defined __APPLE__ && defined __MACH__) && (defined PTHREAD_RECURSIVE_MUTEX_INITIALIZER || defined PTHREAD_RECURSIVE_MUTEX_INITIALIZER_NP)
 
 int
 glthread_recursive_lock_init_multithreaded (gl_recursive_lock_t *lock)
--- gettext-runtime/intl/lock.h.orig	2010-06-06 16:49:57.000000000 +0400
+++ gettext-runtime/intl/lock.h	2012-08-24 20:29:00.000000000 +0400
@@ -272,7 +272,7 @@
 
 # if HAVE_PTHREAD_MUTEX_RECURSIVE
 
-#  if defined PTHREAD_RECURSIVE_MUTEX_INITIALIZER || defined PTHREAD_RECURSIVE_MUTEX_INITIALIZER_NP
+#  if !(defined __APPLE__ && defined __MACH__) && (defined PTHREAD_RECURSIVE_MUTEX_INITIALIZER || defined PTHREAD_RECURSIVE_MUTEX_INITIALIZER_NP)
 
 typedef pthread_mutex_t gl_recursive_lock_t;
 #   define gl_recursive_lock_define(STORAGECLASS, NAME) \
--- gettext-tools/gnulib-lib/glthread/lock.c.orig	2010-05-24 13:42:37.000000000 +0400
+++ gettext-tools/gnulib-lib/glthread/lock.c	2012-08-24 20:29:00.000000000 +0400
@@ -279,7 +279,7 @@
 
 # if HAVE_PTHREAD_MUTEX_RECURSIVE
 
-#  if defined PTHREAD_RECURSIVE_MUTEX_INITIALIZER || defined PTHREAD_RECURSIVE_MUTEX_INITIALIZER_NP
+#  if !(defined __APPLE__ && defined __MACH__) && (defined PTHREAD_RECURSIVE_MUTEX_INITIALIZER || defined PTHREAD_RECURSIVE_MUTEX_INITIALIZER_NP)
 
 int
 glthread_recursive_lock_init_multithreaded (gl_recursive_lock_t *lock)
--- gettext-tools/gnulib-lib/glthread/lock.h.orig	2010-05-24 13:42:37.000000000 +0400
+++ gettext-tools/gnulib-lib/glthread/lock.h	2012-08-24 20:29:00.000000000 +0400
@@ -271,7 +271,7 @@
 
 # if HAVE_PTHREAD_MUTEX_RECURSIVE
 
-#  if defined PTHREAD_RECURSIVE_MUTEX_INITIALIZER || defined PTHREAD_RECURSIVE_MUTEX_INITIALIZER_NP
+#  if !(defined __APPLE__ && defined __MACH__) && (defined PTHREAD_RECURSIVE_MUTEX_INITIALIZER || defined PTHREAD_RECURSIVE_MUTEX_INITIALIZER_NP)
 
 typedef pthread_mutex_t gl_recursive_lock_t;
 #   define gl_recursive_lock_define(STORAGECLASS, NAME) \
--- gettext-tools/libgettextpo/glthread/lock.c.orig	2010-02-17 00:33:14.000000000 +0300
+++ gettext-tools/libgettextpo/glthread/lock.c	2012-08-24 20:29:00.000000000 +0400
@@ -279,7 +279,7 @@
 
 # if HAVE_PTHREAD_MUTEX_RECURSIVE
 
-#  if defined PTHREAD_RECURSIVE_MUTEX_INITIALIZER || defined PTHREAD_RECURSIVE_MUTEX_INITIALIZER_NP
+#  if !(defined __APPLE__ && defined __MACH__) && (defined PTHREAD_RECURSIVE_MUTEX_INITIALIZER || defined PTHREAD_RECURSIVE_MUTEX_INITIALIZER_NP)
 
 int
 glthread_recursive_lock_init_multithreaded (gl_recursive_lock_t *lock)
--- gettext-tools/libgettextpo/glthread/lock.h.orig	2010-03-28 22:33:18.000000000 +0400
+++ gettext-tools/libgettextpo/glthread/lock.h	2012-08-24 20:29:00.000000000 +0400
@@ -271,7 +271,7 @@
 
 # if HAVE_PTHREAD_MUTEX_RECURSIVE
 
-#  if defined PTHREAD_RECURSIVE_MUTEX_INITIALIZER || defined PTHREAD_RECURSIVE_MUTEX_INITIALIZER_NP
+#  if !(defined __APPLE__ && defined __MACH__) && (defined PTHREAD_RECURSIVE_MUTEX_INITIALIZER || defined PTHREAD_RECURSIVE_MUTEX_INITIALIZER_NP)
 
 typedef pthread_mutex_t gl_recursive_lock_t;
 #   define gl_recursive_lock_define(STORAGECLASS, NAME) \
