--- magick/timer.c.orig	2016-09-05 22:20:24.000000000 +0300
+++ magick/timer.c	2017-04-23 13:53:46.000000000 +0300
@@ -129,21 +129,33 @@
 static double ElapsedTime(void)
 {
 #if defined(HAVE_CLOCK_GETTIME)
-  struct timespec timer;
-  (void) clock_gettime(CLOCK_ID, &timer);
+#if defined(__APPLE__)
+  if(clock_gettime != NULL) {
+#endif
+    struct timespec timer;
+    (void) clock_gettime(CLOCK_ID, &timer);
 
-  return((double) timer.tv_sec + timer.tv_nsec/NANOSECONDS_PER_SECOND);
-#elif defined(HAVE_TIMES)
-  struct tms
-    timer;
+    return((double) timer.tv_sec + timer.tv_nsec/NANOSECONDS_PER_SECOND);
+#if defined(__APPLE__)
+  } else {
+#endif
+#endif
+#if !defined(HAVE_CLOCK_GETTIME) || defined(__APPLE__)
+#if defined(HAVE_TIMES)
+    struct tms
+      timer;
 
-  return((double) times(&timer)/CLK_TCK);
+    return((double) times(&timer)/CLK_TCK);
 #else
 #if defined(MSWINDOWS)
-  return(NTElapsedTime());
+    return(NTElapsedTime());
 #else
-  return((double) clock()/CLK_TCK);
+    return((double) clock()/CLK_TCK);
+#endif
+#endif
 #endif
+#if defined(__APPLE__) && defined(HAVE_CLOCK_GETTIME)
+  }
 #endif
 }
 
