--- science/gmic/Portfile.orig	2021-06-23 22:07:46.000000000 +0200
+++ science/gmic/Portfile	2021-06-23 22:09:30.000000000 +0200
@@ -45,7 +45,6 @@
                     size    7401418
 
 depends_lib         port:curl \
-                    path:lib/libavcodec.dylib:ffmpeg \
                     port:fftw-3 \
                     port:GraphicsMagick \
                     port:ilmbase \
@@ -91,6 +90,7 @@
     }
 
     configure.args-append       -DCUSTOM_CFLAGS=ON \
+                                -DENABLE_FFMPEG=OFF \
                                 -DENABLE_OPENMP=OFF \
                                 -DENABLE_X=OFF
 
