--- lib/system/fastopen.c	2017-12-26 18:00:54.041282870 +0100
+++ lib/system/fastopen.c.new	2017-12-26 18:00:09.209281540 +0100
@@ -37,7 +37,7 @@
 #endif
 
 /* TCP Fast Open on OSX behaves differently from Linux, so define these helpers */
-#if defined __APPLE__ && defined __MACH__ && defined CONNECT_DATA_IDEMPOTENT && defined CONNECT_RESUME_ON_READ_WRITE
+#if 0
 # if defined __has_builtin && __has_builtin(__builtin_available)
 #  define TCP_FASTOPEN_OSX
 # endif
--- gl/explicit_bzero.c.orig	2020-08-04 16:59:54.000000000 +0300
+++ gl/explicit_bzero.c	2020-08-04 17:00:24.000000000 +0300
@@ -52,8 +52,6 @@
   (void) SecureZeroMemory (s, len);
 #elif HAVE_EXPLICIT_MEMSET
   explicit_memset (s, '\0', len);
-#elif HAVE_MEMSET_S
-  (void) memset_s (s, len, '\0', len);
 #else
   memset (s, '\0', len);
 # if defined __GNUC__ && !defined __clang__
--- src/gl/explicit_bzero.c.orig	2020-08-04 17:00:42.000000000 +0300
+++ src/gl/explicit_bzero.c	2020-08-04 17:00:53.000000000 +0300
@@ -52,8 +52,6 @@
   (void) SecureZeroMemory (s, len);
 #elif HAVE_EXPLICIT_MEMSET
   explicit_memset (s, '\0', len);
-#elif HAVE_MEMSET_S
-  (void) memset_s (s, len, '\0', len);
 #else
   memset (s, '\0', len);
 # if defined __GNUC__ && !defined __clang__
