--- src/gtkosxapplication_quartz.c.orig	2021-10-16 17:36:03.000000000 +0200
+++ src/gtkosxapplication_quartz.c	2021-10-16 18:00:42.000000000 +0200
@@ -593,6 +593,7 @@
   [NSApp setDelegate: [GtkApplicationDelegate new]];
   self->priv->delegate = [NSApp delegate];
   bindtextdomain (PACKAGE_NAME, LOCALEDIR);
+  bind_textdomain_codeset(PACKAGE_NAME, "UTF-8");
 
   /* Check if we're running inside an application bundle and overwrite the
    * previously bound domain to a location inside the bundle.
