--- coders/Makefile.am.orig	2018-12-05 18:48:02.000000000 +0300
+++ coders/Makefile.am	2018-12-05 18:51:28.000000000 +0300
@@ -534,7 +534,7 @@
 coders_fpx_la_SOURCES      = coders/fpx.c
 coders_fpx_la_CPPFLAGS     = $(MAGICK_CODER_CPPFLAGS)
 coders_fpx_la_LDFLAGS      = $(MODULECOMMONFLAGS)
-coders_fpx_la_LIBADD       = $(MAGICKCORE_LIBS) $(FPX_LIBS) $(MATH_LIBS) -lstdc++
+coders_fpx_la_LIBADD       = $(MAGICKCORE_LIBS) $(FPX_LIBS) $(MATH_LIBS)
 
 # GIF coder module
 coders_gif_la_SOURCES      = coders/gif.c
