--- graphics/librsvg/Portfile.orig	2019-10-19 14:14:01.000000000 +0300
+++ graphics/librsvg/Portfile	2019-10-19 14:14:48.000000000 +0300
@@ -55,7 +55,7 @@
 # rust @1.30.1 fails to build on 10.10 and earlier
 # https://trac.macports.org/ticket/57768
 
-if {${os.platform} eq "darwin" && ${os.major} < 15} {
+if {${os.platform} eq "darwin"} {
     # revert to latest pre-cargo version
     version         2.40.20
     revision        3
