--- pango/pangocoretext-fontmap.c.orig	2017-08-16 00:53:10.000000000 +0300
+++ pango/pangocoretext-fontmap.c	2017-10-19 11:18:30.000000000 +0300
@@ -827,6 +827,12 @@
 {
   double size = pango_font_description_get_size (desc);
 
+  if (!pango_font_description_get_size_is_absolute (desc))
+    {
+      double dpi = pango_core_text_font_map_get_resolution (fontmap, context);
+      size = size * dpi / 72.;
+    }
+
   return .5 + pango_matrix_get_font_scale_factor (pango_context_get_matrix (context)) * size;
 }
 
