--- configure.orig	2019-11-09 14:40:46.000000000 +0300
+++ configure	2019-11-09 14:41:12.000000000 +0300
@@ -9528,7 +9528,7 @@
 		# to ensure that tests don't crash
 		# Note: This matches the value of THREAD_STACK_SIZE in
 		# thread_pthread.h
-		LINKFORSHARED="-Wl,-stack_size,1000000 $LINKFORSHARED"
+		#LINKFORSHARED="-Wl,-stack_size,1000000 $LINKFORSHARED"
 
 		if test "$enable_framework"
 		then
